ibis/backends/tests/test_map.py

```diff
@@ -232,19 +317,43 @@ def test_map_get_with_null_on_null_type_with_null(con, null_value):
 
 
 @pytest.mark.notyet(["postgres"], reason="only support maps of string -> string")
+@pytest.mark.notimpl(
+    ["flink"],
+    raises=NotImplementedError,
+    reason="No translation rule for map<string, null>",
+)
 def test_map_get_with_null_on_null_type_with_non_null(con):
     value = ibis.literal({"A": None, "B": None})
     expr = value.get("C", 1)
     assert con.execute(expr) == 1
 
 
+@pytest.mark.notimpl(
+    ["flink"],
+    raises=exc.UnsupportedBackendType,
+    reason="UnsupportedBackendType: map",
+)
 def test_map_create_table(con, temp_table):
-    t = con.create_table(
-        temp_table, schema=ibis.schema(dict(xyz="map<string, string>"))
-    )
+    if con.name == "flink":
+        t = con.create_table(
+            temp_table,
+            schema=ibis.schema(dict(xyz="map<string, string>")),
+            tbl_properties={
+                "connector": None,
+            },
```

Comment:
@mfatihaktas what's this? haven't looked into it, but just curious to see a different interface required

Reply:
This was "my workaround" for the following error raised in `create_table`:
```python
        import pandas as pd
        import pyarrow as pa

        import ibis.expr.types as ir

        if obj is None and schema is None:
            raise exc.IbisError("The schema or obj parameter is required")

        if overwrite:
            self.drop_table(name=name, catalog=catalog, database=database, force=True)

        if isinstance(obj, pa.Table):
            obj = obj.to_pandas()
        if isinstance(obj, pd.DataFrame):
            qualified_name = self._fully_qualified_name(name, database, catalog)
            table = self._table_env.from_pandas(obj)
            # in-memory data is created as views in `pyflink`
            # TODO(chloeh13q): alternatively, we can do CREATE TABLE and then INSERT
            # INTO ... VALUES to keep things consistent
            self._table_env.create_temporary_view(qualified_name, table)
        if isinstance(obj, ir.Table):
            # TODO(chloeh13q): implement CREATE TABLE for expressions
            raise NotImplementedError
        if schema is not None:
            if not tbl_properties:
>               raise exc.IbisError(
                    "tbl_properties is required when creating table with schema"
                )
E               ibis.common.exceptions.IbisError: tbl_properties is required when creating table with schema
```

I did not know how I should modify `create_table()`, so chose to work around it this way :) Given that it works with
```
    tbl_properties={
                "connector": None,
            },
```
which is technically "empty", this assert
```
        if schema is not None:
            if not tbl_properties:
>               raise exc.IbisError(
                    "tbl_properties is required when creating table with schema"
                )
```
is unnecessary perhaps?

Reply:
@chloeh13q thoughts? Doesn't seem like passing `"connector": None` and getting

```
CREATE TABLE...
WITH(
    'connector' = None
)
```

doesn't seem it should be right

Reply:
I just tested this in Flink and I think it's technically possible to create tables without table properties, i.e.
```sql
CREATE TABLE MyTable (
  `user_id` BIGINT,
  `item_id` BIGINT,
  `behavior` STRING
);
```
This would assume a managed table (where the connection information is already filled in the session environment; as opposed to an external table) & would throw `org.apache.flink.table.api.ValidationException` if such managed table is not found. Therefore, I think we can remove this validation & the exception.

That being said, we can also mark this test as not supported / not implemented in Flink backend and create a separate test somewhere else testing this functionality (given the slight difference in expected behavior as compared to other backends). That's what we did with other tests around `create_table()`: https://github.com/ibis-project/ibis/blob/master/ibis/backends/flink/tests/test_ddl.py .

Reply:
I would prefer to remove this validation, if that can allow for a more uniform API. There are several instances of this, and would like to avoid having to replicate a lot of tests where it doesn't seem necessary (i.e. it looks like just dropping the difference would be fine).

Reply:
Will address this in a follow-up PR, as it will require modifying `create_table` (don't want that side effect to get conflated with this PR).

Reply:
sounds reasonable to me